idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4522 @@
+        ax3.axvline(t_heating[-1], lw=0.7, color="r", ls="--")
+        ax3.axvline(t_cooling[-1], lw=0.7, color="r", ls="--")
+        ax3.axvline(t_pressurization[-1], lw=0.7, color="r", ls="--")
+        ax3.axvline(t_adsorption[-1], lw=0.7, color="r", ls="--")
+        ax3.grid()
+
+        plt.show()
+
+    def _var_dict(self):
+
+        var_dict = {}
+
+        var_dict["Adsorption temperature [K]"] = self.temperature_adsorption
+        var_dict["Desorption temperature [K]"] = self.temperature_desorption
+        var_dict["Heating temperature [K]"] = self.temperature_heating
+        var_dict["Cooling temperature [K]"] = self.temperature_cooling
+        var_dict["Column diameter [m]"] = self.bed_diameter
+        var_dict["Column length [m]"] = self.bed_height
+        var_dict["Column volume [m3]"] = self.bed_volume
+        var_dict[
+            "CO2 mole fraction at feed [%]"] = self.mole_frac_in["CO2"] * 100
+        var_dict["Feed flow rate [mol/s]"] = self.flow_mol_in_total
+        var_dict["Feed velocity [m/s]"] = self.velocity_in
+        var_dict["Minimum fluidization velocity [m/s]"] = self.velocity_mf
+
+        var_dict["Time of heating step [h]"] = units.convert(
+            self.heating.time, to_units=units.hr)
+        var_dict["Time of cooling step [h]"] = units.convert(
+            self.cooling.time, to_units=units.hr)
+        var_dict["Time of pressurization step [h]"] = units.convert(
+            self.pressurization.time, to_units=units.hr)
+        var_dict["Time of adsorption step [h]"] = units.convert(
+            self.adsorption.time, to_units=units.hr)
+        var_dict["Cycle time [h]"] = self.cycle_time
+
+        var_dict["Purity [-]"] = self.purity
+        var_dict["Recovery [-]"] = self.recovery
+        var_dict["Productivity [kg CO2/ton/h]"] = self.productivity
+        var_dict["Specific energy [MJ/kg CO2]"] = self.specific_energy
+        var_dict["Heat duty per bed [MW]"] = self.heat_duty_bed
+        var_dict["Heat duty total [MW]"] = self.heat_duty_total
+
+        var_dict["Pressure drop [Pa]"] = -self.pressure_drop
+        var_dict["Number of beds"] = self.number_beds
+
+        var_dict["CO2 captured in one cycle per bed [kg/cycle]"] = (
+            self.CO2_captured_bed_cycle)
+
+        var_dict["Cycles per year"] = self.cycles_year
+
+        var_dict[
+            "Total CO2 captured per year [tonne/year]"
+        ] = self.total_CO2_captured_year
+
+        var_dict[
+            "Amount of flue gas processed per year [Gmol/year]"
+        ] = self.flue_gas_processed_year
+
+        var_dict[
+            "Amount of flue gas processed per year (target) [Gmol/year]"
+        ] = self.flue_gas_processed_year_target
+
+        var_dict["Amount of CO2 to atmosphere [mol/s]"] = self.emissions_co2
+
+        var_dict[
+            "Concentration of CO2 emitted to atmosphere [ppm]"
+        ] = self.emissions_co2_ppm
+
+        return var_dict
+
+    def summary(self, export=False):
```

Comment:
This looks like it might be something that should go into a utility function file, rather than part of the unit model itself.

Reply:
moved to a util file